homework/homework5/node.py

```diff
@@ -0,0 +1,173 @@
+                and self.left_child == other.left_child
+                and self.right_child == other.right_child
+            )
+        return False
+
+    def __contains__(self, key) -> bool:
+        if self.key == key:
+            return True
+
+        if (self.left_child is not None) and (key < self.key):
+            return key in self.left_child
+
+        if (self.right_child is not None) and (key > self.key):
+            return key in self.right_child
+
+        return False
+
+    def __iter__(self):
+        yield self.value
+        if self.left_child is not None:
+            yield from self.left_child
+        if self.right_child is not None:
+            yield from self.right_child
+
+    def __reversed__(self):
+        if self.left_child is not None:
+            yield from reversed(self.left_child)
+        if self.right_child is not None:
+            yield from reversed(self.right_child)
+        yield self.value
+
+    def insert(self, key, value, priority: Optional[float] = None) -> "Node":
+        new_node = Node(key, value, priority)
+
+        left_split, right_split = self.__split(key)
+
+        if left_split is None:
+            return new_node.__merge(right_split)
+        else:
+            left_split = left_split.__merge(new_node)
+            return left_split.__merge(right_split)
+
+    def get(self, key) -> Any:
+        if self.key == key:
+            return self.value
+
+        if (self.left_child is not None) and (key < self.key):
+            return self.left_child.get(key)
+
+        if (self.right_child is not None) and (key > self.key):
+            return self.right_child.get(key)
+
+    def update(self, key, new_value):
+        if self.key == key:
+            self.value = new_value
+
+        if (self.left_child is not None) and (key < self.key):
+            self.left_child.update(key, new_value)
+
+        if (self.right_child is not None) and (key > self.key):
+            self.right_child.update(key, new_value)
+
+    def remove(self, key) -> Optional["Node"]:
+        left_subtree, right_subtree = self.__split(key)
+
+        if right_subtree is None:
+            return left_subtree
+        else:
+            right_subtree = right_subtree.__remove_smallest()
+
+        if left_subtree is None:
+            return right_subtree
+
+        return left_subtree.__merge(right_subtree)
+
+    def __remove_smallest(self) -> Optional["Node"]:
+        if self.left_child is None:
+            return self.right_child
+
+        self.left_child = self.left_child.__remove_smallest()
+        return self
+
+    def __split(self, key) -> Tuple[Optional["Node"], Optional["Node"]]:
+        if key > self.key:
+            if self.right_child is None:
+                return self, None
+
+            left_subtree, right_subtree = self.right_child.__split(key)
+
+            result = self
+            result.right_child = left_subtree
+            return result, right_subtree
+        else:
```

Comment:
You can delete else branch and use code directly since the if branch has return statment